src/app/base/components/AppSideNavigation/AppSideNavigation.tsx

```diff
@@ -0,0 +1,410 @@
+import { actions as controllerActions } from "app/store/controller";
+import controllerSelectors from "app/store/controller/selectors";
+import { version as versionSelectors } from "app/store/general/selectors";
+import type { RootState } from "app/store/root/types";
+import { actions as statusActions } from "app/store/status";
+import type { User } from "app/store/user/types";
+
+const navGroups: NavGroup[] = [
+  {
+    navLinks: [
+      {
+        label: "Images",
+        url: urls.images.index,
+      },
+      {
+        highlight: [urls.tags.index, urls.tags.tag.index(null)],
+        label: "Tags",
+        url: urls.tags.index,
+      },
+      {
+        highlight: [urls.zones.index, urls.zones.details(null)],
+        label: "AZs",
+        url: urls.zones.index,
+      },
+      {
+        label: "Pools",
+        url: urls.pools.index,
+      },
+    ],
+  },
+  {
+    groupTitle: "Hardware",
+    groupIcon: "machines-light",
+    navLinks: [
+      {
+        highlight: [urls.machines.index, urls.machines.machine.index(null)],
+        label: "Machines",
+        url: urls.machines.index,
+      },
+      {
+        highlight: [urls.devices.index, urls.devices.device.index(null)],
+        label: "Devices",
+        url: urls.devices.index,
+      },
+      {
+        adminOnly: true,
+        highlight: [
+          urls.controllers.index,
+          urls.controllers.controller.index(null),
+        ],
+        label: "Controllers",
+        url: urls.controllers.index,
+      },
+    ],
+  },
+  {
+    groupTitle: "KVM",
+    groupIcon: "cluster-light",
+    navLinks: [
+      {
+        label: "LXD",
+        url: urls.kvm.lxd.index,
+      },
+      {
+        label: "Virsh",
+        url: urls.kvm.virsh.index,
+      },
+    ],
+  },
+  {
+    groupTitle: "Networking",
+    groupIcon: "connected-light",
+    navLinks: [
+      {
+        highlight: [
+          urls.subnets.index,
+          urls.subnets.subnet.index(null),
+          urls.subnets.space.index(null),
+          urls.subnets.fabric.index(null),
+          urls.subnets.vlan.index(null),
+        ],
+        label: "Subnets",
+        url: urls.subnets.index,
+      },
+      {
+        highlight: [urls.domains.index, urls.domains.details(null)],
+        label: "DNS",
+        url: urls.domains.index,
+      },
+      {
+        label: "Network discovery",
+        url: urls.dashboard.index,
+      },
+    ],
+  },
+];
+
+const generateItems = ({
+  authUser,
+  groups,
+  isAdmin,
+  isAuthenticated,
+  logout,
+  path,
+  showLinks,
+  vaultIncomplete,
+}: {
+  authUser: User | null;
+  groups: NavGroup[];
+  isAdmin: boolean;
+  isAuthenticated: boolean;
+  logout: () => void;
+  path: string;
+  showLinks: boolean;
+  vaultIncomplete: boolean;
+}) => {
+  const items = [];
+
+  if (showLinks) {
+    groups.forEach((group) => {
+      items.push(
+        <>
+          <div className="p-muted-heading" id={group.groupTitle}>
```

Comment:
I can see some problems with this approach, `id` needs to be unique and group title is very generic. One way of solving this:

```jsx
const id = useId();
<div className="p-muted-heading" id={`${group.groupTitle}-${id}}>
```